semgrep-core/bin/Main.ml

```diff
@@ -264,6 +275,55 @@ let gen_layer ~root ~query file =
   ()
 (*e: function [[Main_semgrep_core.gen_layer]] *)
 
+let filemtime file =
+  (Unix.stat file).Unix.st_mtime
+
+(* The function below is mostly a copy-paste of Common.cache_computation.
+ * This function is slightly more flexible because we can put the cache file
+ * anywhere thanks to the argument 'cache_file_of_file'.
+ * We also try to be a bit more type-safe by using the version tag above.
+ * TODO: merge in pfff/commons/Common.ml at some point
+ *)
+let cache_computation file cache_file_of_file f =
+  if not !use_parsing_cache
+  then f ()
+  else begin
+    if not (Sys.file_exists file)
```

Comment:
What are your thoughts re. adding the file md5 checksum here? This would enable the Python wrapper to skip calculating this value itself.

Reply:
we don't compute the md5sum of the file content, just of the filename

Reply:
The suggestion here is that we do calculate the md5 sum of the file content. Given that we're already lexing the entire file, this seems relatively cheap for us here, vs. done in the Python wrapper.